File: SampleRun.m
% Refer to AROPE.m for details

% A sample run is as follows:

edge_list = load('BlogCatalog.csv');
A = sparse(edge_list(:,1),edge_list(:,2),1,max(max(edge_list)),max(max(edge_list)));
A = A + A';
order = [1,2,3,-1];
weights = cell(4,1);
weights{1} = 1;
weights{2} = [1,0.1];
weights{3} = [1,0.1,0.01];
weights{4} = 0.001;
[U_cell,V_cell] = AROPE(A,128,order,weights);
% Network Reconstruction
for i = 1:4
    results = Precision_Np(A,sparse(max(max(edge_list)),max(max(edge_list))),U_cell{i},V_cell{i},1e6);
    figure(i);
    semilogx(1:1e6,results);
end

File: python/Sample_Run.py
# Sample run on BlogCatalog
import numpy as np
import pandas as pd
from scipy.sparse import csr_matrix

import utils
from eval import Precision_Np

if __name__ == '__main__':

    data = pd.read_csv('BlogCatalog.csv')      
    data = np.array(data) - 1                       # change index from 0
    N = np.max(np.max(data)) + 1
    A = csr_matrix((np.ones(data.shape[0]), (data[:,0],data[:,1])), shape = (N,N))
    A += A.T

    order = [1,2,3,-1]
    weights = []
    weights.append([1])
    weights.append([1,0.1])
    weights.append([1,0.1,0.01])
    weights.append([0.001])
    U_list,V_list = utils.AROPE(A,128,order,weights)
    # Network Reconstruction
    results = [Precision_Np(A,csr_matrix((N,N)),U_list[i],V_list[i],1e6) for i in range(4)]
    
    